packages/utils/src/deviceDetector/types.ts

```diff
@@ -0,0 +1,41 @@
+export interface UADataValues {
+  platform: string;
+  platformVersion: string;
+  architecture: string;
+  model: string;
+  uaFullVersion: string;
+}
+
+export interface AgentBrowserInfo {
+  name: string;
+  version: string;
+}
+export interface AgentInfo {
+  isMobile: boolean;
+  isDesktop: boolean;
+  browser: AgentBrowserInfo;
+}
+
+export interface NavigatorUABrandVersion {
+  brand: string;
+  version: string;
+}
+
+export interface PresetInfo {
+  test: string;
+  id: string;
+}
+
+export interface PresetResult {
+  preset: PresetInfo | null;
+  version: string;
+}
+
+export interface NavigatorUAData {
+  brands: NavigatorUABrandVersion[];
+  uaList?: NavigatorUABrandVersion[];
+  mobile: boolean;
+  getHighEntropyValues<T extends keyof UADataValues>(
+    hints: T[]
+  ): Promise<{ [key in T]: UADataValues[T] }>;
```

Comment:
요 친구는 메서드 방식으로 선언한 이유가 있나유? 메서드 방식으로 선언하면 함수의 인자가 이변적으로 평가되게 되는데, 함수의 인자가 이변적으로 평가되는 것을 이용하는 상황은 거의 없어서용.

```suggestion
  getHighEntropyValues: <T extends keyof string>(
    hints: T[]
  ) => Promise<{ [key in T]: string[T] }>;
```

이런 경우는 이렇게 프로퍼티 방식으로 선언하여 함수의 인자가 반공변적으로 평가되도록 만드는 것이 더 타입 안정성을 높힐 수 있는 방법입니다...!

공변성, 반공변성, 이변성에 대한 내용은 [이 포스팅](https://seob.dev/posts/%EA%B3%B5%EB%B3%80%EC%84%B1%EC%9D%B4%EB%9E%80-%EB%AC%B4%EC%97%87%EC%9D%B8%EA%B0%80/) 한번 읽어보시면 좋을 것 같아요...!!